packages/user-management/src/standard-user-workflow/legacy/UserLegacyAdapter.ts

```diff
@@ -195,21 +198,13 @@ export class LegacyAdapterUserRepository implements Bootstrappable, UserReposito
     try {
       const login = await this.getLoginFromAccountId(accountId);
 
-      const result = await this.simple.setStatus({
+      await this.simple.setStatus({
         status: {
-          active: newStatus === StandardAccountStatus.Active
+          active: newStatus === StandardAccountStatus.Active,
+          data: newStatus
         },
         key: login
       });
-
-      if (result.fields) {
-        await this.simple.updateAccount({
-          key: result.fields.login,
-          fields: {
-            accountStatus: newStatus
-          }
-        });
-      }
     } catch (e) {
       throw new LegacyUpdateAccountStatus(`Failed to update the status of the user (accountId: ${accountId}.`, e);
```

Comment:
There is a missing parenthesis. For better encapsulation and to be able to get useful information, accountId should also be stored in the exception